File: tests/human/x/cs/short_circuit.cs
using System;

class Program
{
    static bool Boom(int a, int b)
    {
        Console.WriteLine("boom");
        return true;
    }

    static void Main()
    {
        Console.WriteLine(false && Boom(1, 2));
        Console.WriteLine(true || Boom(1, 2));
    }
}

File: tests/human/x/pas/short_circuit.pas
program ShortCircuit;

function boom(a, b: integer): boolean;
begin
  Writeln('boom');
  boom := True;
end;

var
  res: boolean;
begin
  if False then
    res := boom(1,2)
  else
    res := False;
  Writeln(res);

  if True then
    res := True
  else
    res := boom(1,2);
  Writeln(res);
end.
